src/swupd/files.go

```diff
@@ -0,0 +1,130 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+)
+
+// File represents an entry in a manifest
+type File struct {
+	Name        string
+	Hash        string
+	Version     uint32
+
+	// flags
+	Type        string
+	Status      string
+	Modifier    string
+	Rename      bool
+
+	// renames
+	RenameScore uint16
+	RenamePeer  *File
+
+	Info        os.FileInfo
+	DeltaPeer   *File
+}
+
+// set file type based on flag
+// F: file
+// D: directory
+// L: link
+// M: manifest
+// .: none (may be deleted/ghosted)
+func setFileTypeFromFlag(flag string, f *File) error {
```

Comment:
If the flag is a character we should say so, rather than pretending it is a string.